doc/subsystems/bluetooth/hci.txt

```diff
@@ -0,0 +1,994 @@
++--------------------------+-------+--------------------+--------------------+
+| Read_Host_Stack_Commands | 0x00C |                    | Status,            |
+|                          |       |                    | Num_Commands,      |
+|                          |       |                    | Vendor_ID[i],      |
+|                          |       |                    | Opcode_Base[i]     |
++--------------------------+-------+--------------------+--------------------+
+
+This command shall return the list of host stack operation defined vendor
+commands. They are defined based on a vendor identifier and a base opcode.
+
+With this command it is possible to incorporate external defined HCI commands
+designed for host stack operation that extend the standard and vendor specific
+Controller commands. See Android [1][2] and Microsoft [3] defined extensions.
+
+The Opcode_Base parameter points to an opcode that will identify the external
+command set. Ideally all external commands are multiplexed through that single
+opcode and also define a flexible interface for external events.
+
+	Status:                                         Size: 1 Octet
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0x00               | Command succeeded                    |
+	| 0x01-0xFF          | Command failed                       |
+	+--------------------+--------------------------------------+
+
+	Num_Commands:                                   Size: 1 Octet
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0xXX               | Total number of commands             |
+	+--------------------+--------------------------------------+
+
+	Vendor_ID[i]:                                  Size: 2 Octets
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0x0001             | Android specific HCI commands        |
+	| 0x0002             | Microsoft specific HCI commands      |
+	| All other values   | Reserved for future use              |
+	+--------------------+--------------------------------------+
+
+	Opcode_Base[i]:                                Size: 2 Octets
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0xXXXX             | Base opcode of command               |
+	+--------------------+--------------------------------------+
+
+When the Read_Host_Stack_Commands command has completed, a Command Complete
+event shall be generated.
+
+
+Zephyr Set Scan Request Reports Command
```

Comment:
That was the idea. If we make advertising extensions an optional feature, then we can also keep its extended HCI commands optional. Then this could result in a lot smaller binaries.